src/vaults/vaults.service.ts

```diff
@@ -33,6 +52,103 @@ export class VaultsService {
     return VaultsService.loadVault(chain, vaultDefinition, currency);
   }
 
+  async listVaultHarvests(chain: Chain): Promise<VaultHarvestsMap> {
+    const harvestsWithSnapshots = await Promise.all(
+      chain.vaults.map(async (vault) => {
+        return {
+          vault: vault.vaultToken,
+          harvests: await this.getVaultHarvests(chain, vault.vaultToken),
+        };
+      }),
+    );
+
+    return harvestsWithSnapshots.reduce((acc, harvestWithSnapshot) => {
+      acc[harvestWithSnapshot.vault] = harvestWithSnapshot.harvests;
+      return acc;
+    }, <VaultHarvestsMap>{});
+  }
+
+  async getVaultHarvests(chain: Chain, vaultAddr: VaultDefinition['vaultToken']): Promise<VaultHarvestsExtended[]> {
+    const vaultHarvests: VaultHarvestsExtended[] = [];
+
+    vaultAddr = ethers.utils.getAddress(vaultAddr.toLowerCase());
+
+    const sdk = await chain.getSdk();
+    const vaultDef = chain.vaults.find((vault) => vault.vaultToken === vaultAddr);
+
+    let sdkVaultHarvestsResp: {
+      data: VaultHarvestData[];
+    } = { data: [] };
+
+    if (!vaultDef) return vaultHarvests;
+
+    try {
+      sdkVaultHarvestsResp = await sdk.vaults.listHarvests({
+        address: vaultAddr,
+        version: vaultDef.version ?? VaultVersion.v1,
+      });
+    } catch (e) {
+      console.warn(`Failed to get harvests list ${e}`);
+    }
+
+    if (!sdkVaultHarvestsResp || sdkVaultHarvestsResp?.data?.length === 0) {
+      return vaultHarvests;
+    }
+
+    const sdkVaultHarvests = sdkVaultHarvestsResp.data;
+
+    const _extend_harvests_data = async (harvestsList: VaultPerformanceEvent[], eventType: HarvestType) => {
+      if (!harvestsList || harvestsList?.length === 0) return;
+
+      for (let i = 0; i < harvestsList.length; i++) {
+        const harvest = harvestsList[i];
+
+        const vaultGraph = await sdk.graph.loadSett({
```

Comment:
this could get expensive in sequence. we either need to parallelize this call or probably should consider an extension on this pr where we create a ddb table we update running these same methods on a cron. 

ideally we want all our api like sub 200-300ms

Reply:
Sgmt, here is the ticket https://github.com/Badger-Finance/badger-api/issues/1086